lib/base/logger.hpp

```diff
@@ -118,6 +118,16 @@ class Log
 
 	~Log();
 
+	/**
+	 * Set the severity of the current Log instance.
+	 *
+	 * @param severity LogSeverity The new severity you want to override the current one with.
+	 */
+	void SetSeverity(LogSeverity severity)
+	{
+		m_Severity = severity;
+	}
```

Comment:
This won't work properly as the constructor already sets `m_IsNoOp` based on the severity passed to it:

https://github.com/Icinga/icinga2/blob/f16691545ec0400b7f4ea46c7b61395a620aef04/lib/base/logger.cpp#L254-L256

So when changing the severity at a later point, the message may still be discarded incorrectly.